crypto/ct/ct_sct.c

```diff
@@ -317,12 +317,11 @@ int SCT_set_source(SCT *sct, sct_source_t source)
 
 int SCT_LIST_set_source(const STACK_OF(SCT) *scts, sct_source_t source)
 {
-    int i, ret = 1;
+    int i, ret = 0;
 
     for (i = 0; i < sk_SCT_num(scts); ++i) {
-        ret = SCT_set_source(sk_SCT_value(scts, i), source);
-        if (ret != 1)
-            break;
+        if (SCT_set_source(sk_SCT_value(scts, i), source))
```

Comment:
`SCT_LIST_set0_logs` returns the number of SCTs it set the log for, so the caller can tell if it did anything. However, in the cases I can foresee, the caller only cares that any SCTs whose log can be identified, has been. If that's none of them, so be it. The typical next step would be to do something for each SCT where sct->log != NULL (e.g. print the log name, validate the SCT) and just skip/fail for the other SCTs.

As with `SCT_LIST_set_source`, it really is just a convenience function - a caller could achieve the same thing. I'm fine with removing it to slim down the public API.